src/Utils/Hooks/useCountdownTimer.ts

```diff
@@ -1,26 +1,50 @@
 import { useState, useEffect } from "react"
-import { differenceInHours, differenceInMinutes } from "date-fns"
+import {
+  differenceInHours,
+  differenceInMinutes,
+  differenceInSeconds,
+} from "date-fns"
 
 const IMMINENT_TIME = 5 // hours
 
-const calculateTime = (endTime: string) => {
+const calculateTime = (endTime: string, includeSeconds: boolean) => {
   const now = new Date()
   const expiration = new Date(endTime)
-  const timeDiff = differenceInMinutes(expiration, now)
 
-  const ONE_DAY = 1440 // 24 hours * 60 minutes
-
-  const days = Math.floor(timeDiff / ONE_DAY)
-  const hours = Math.floor((timeDiff % ONE_DAY) / 60)
-  const minutes = Math.floor(timeDiff % 60)
-
-  if (timeDiff <= 0) {
-    return "Expired"
-  }
-  if (timeDiff >= ONE_DAY) {
-    return `${days}d ${hours}h`
+  if (!includeSeconds) {
+    const timeDiff = differenceInMinutes(expiration, now)
+
+    const ONE_DAY = 1440 // 24 hours * 60 minutes
+
+    const days = Math.floor(timeDiff / ONE_DAY)
+    const hours = Math.floor((timeDiff % ONE_DAY) / 60)
+    const minutes = Math.floor(timeDiff % 60)
+
+    if (timeDiff <= 0) {
+      return "Expired"
+    }
+    if (timeDiff >= ONE_DAY) {
+      return `${days}d ${hours}h`
+    }
+    return `${hours}h ${minutes}m`
+  } else {
+    const timeDiff = differenceInSeconds(expiration, now)
+
+    const ONE_DAY = 86400 // 24 hours * 60 minutes * 60 seconds
+
+    const days = Math.floor(timeDiff / ONE_DAY)
+    const hours = Math.floor((timeDiff % ONE_DAY) / (60 * 60))
+    const minutes = Math.floor((timeDiff % (60 * 60)) / 60)
+    const seconds = Math.floor(timeDiff % 60)
+
+    if (timeDiff <= 0) {
+      return "Expired"
+    }
+    if (timeDiff >= ONE_DAY) {
+      return `${days}d ${hours}h ${minutes}m ${seconds}s`
```

Comment:
Do we want to include the minutes and seconds if the time difference is >= ONE_DAY? If so, I think we can consolidate the logic in this method by having an `includeMinutes` flag that ensures minutes are always included.

Reply:
That's a good point. I'm going to ask Vivi if we want to show the minutes and seconds when the time difference is greater than ONE_DAY.